controller.go

```diff
@@ -253,14 +302,30 @@ func (c *controller) NewNetwork(networkType, name string, options ...NetworkOpti
 
 	network.processOptions(options...)
 
-	if err := c.addNetwork(network); err != nil {
+	if _, err := c.loadNetworkDriver(network); err != nil {
```

Comment:
loadNetworkDriver is being called again from addNetwork. 
Do we really have to load the plugin twice in the same NewNetwork call ?